From 2e0fb52187504fad6657c9462ce650a540d5e387 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Thu, 21 Mar 2013 23:02:37 +0000
Subject: [PATCH 1/2] BZ#11120: fix x86_64/strcmp.S NOT_IN_libc safeguards

Due to a typo repeated several times, this bug hasn't been fixed yet,
despite being marked as resolved in glibc 2.12.

* sysdeps/x86_64/strcmp.S: Replace all occurrences of NOT_IN_lib
with NOT_IN_libc.
---
 sysdeps/x86_64/strcmp.S |  6 +++---
 3 files changed, 17 insertions(+), 11 deletions(-)

diff --git a/sysdeps/x86_64/strcmp.S b/sysdeps/x86_64/strcmp.S
index de9ecc9..7680937 100644
--- a/sysdeps/x86_64/strcmp.S
+++ b/sysdeps/x86_64/strcmp.S
@@ -32,7 +32,7 @@
 /* The simplified code below is not set up to handle strncmp() so far.
    Should this become necessary it has to be implemented.  For now
    just report the problem.  */
-# ifdef NOT_IN_lib
+# ifdef NOT_IN_libc
 #  error "strncmp not implemented so far"
 # endif
 
@@ -51,7 +51,7 @@
 # include "locale-defines.h"
 
 /* No support for strcasecmp outside libc so far since it is not needed.  */
-# ifdef NOT_IN_lib
+# ifdef NOT_IN_libc
 #  error "strcasecmp_l not implemented so far"
 # endif
 
@@ -60,7 +60,7 @@
 # include "locale-defines.h"
 
 /* No support for strncasecmp outside libc so far since it is not needed.  */
-# ifdef NOT_IN_lib
+# ifdef NOT_IN_libc
 #  error "strncasecmp_l not implemented so far"
 # endif
 
-- 
1.8.2

